python/pyrogue/_Root.py

```diff
@@ -696,9 +695,12 @@ def keyValueUpdate(old, key, value):
     d = old
     parts = key.split('.')
     for part in parts[:-1]:
-        d = d.get(part, {})
+        if part not in d:
```

Comment:
Does the default return not work properly?

Reply:
It needs to return a new dict, but also push that new dict into `d` at `part` before assigning over `d`. `d.get(part, {})` doesn't do the push.
What i was seeing was that if `keyValueUpdate()` was called with an empty top level dict, it would not actually fill it with anything.